File: src/test/tls.c
/* -*- Mode: C; tab-width: 8; c-basic-offset: 2; indent-tabs-mode: nil; -*- */

#include "util.h"

__thread int tlsvar;

void breakpoint_fn(void) {}

void* thread_fn(void* arg) {
  tlsvar = *(int*)arg;
  breakpoint_fn();
  return NULL;
}

int main(void) {
  pthread_t thread;

  int value = 97;
  pthread_create(&thread, NULL, thread_fn, &value);
  pthread_join(thread, NULL);

  return 0;
}

File: src/test/tls.py
from util import *

send_gdb('break breakpoint_fn')
expect_gdb('Breakpoint 1')

send_gdb('c')
expect_gdb('Breakpoint 1')
send_gdb('print tlsvar')
expect_gdb(' = 97')

send_gdb('reverse-stepi')
send_gdb('print tlsvar')
expect_gdb(' = 97')

ok()
